### PR TITLE
Validator app support

### PR DESCRIPTION
**Pull request by raynaudoe**
On this PR:
-  Added necessary code to manage both app modes (Consumer, Validator)
-  logging for ledger device interactions

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -35,16 +39,56 @@ const (
...
+		return ListingDerivationPath
+	}
+
+	return []uint32{}
+}
+
+
 // Displays existing Ledger Oasis apps by address
-func ListOasisDevices(path []uint32) {
+func ListOasisDevices(mode LedgerAppMode) {
```
Comment (jleni) [inline app.go:87]: This change can be a problem:
- in some cases, apps may want to try user range of derivation paths which are all valid for the specific mode. 
We should not limit this to only two options. 
I think a good option is to use the two first items in the derivation path to define the mode or reject the call.
- it breaks API compatibility with the existing Oasis CLI
-

File: app.go [hunk 2]
```diff
@@ -72,7 +116,7 @@ func ListOasisDevices(path []uint32) {
 }
 
 // ConnectLedgerOasisApp connects to Oasis app based on address
-func ConnectLedgerOasisApp(seekingAddress string, path []uint32) (*LedgerOasis, error) {
+func ConnectLedgerOasisApp(seekingAddress string, mode LedgerAppMode) (*LedgerOasis, error) {
```
Comment (jleni) [inline app.go:119]: Same as previous comment

File: app.go (hunk 1 above)
Comment (raynaudoe) [reply app.go]: ok, fixed!

File: app.go (hunk 2 above)
Comment (raynaudoe) [reply app.go]: also fixed.

**Pull request merged**